hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -4050,7 +4052,12 @@ private void commitOrCompleteLastBlock(
   private void finalizeINodeFileUnderConstruction(String src,
       INodeFile pendingFile)
       throws IOException {
-    finalizeINodeFileUnderConstructionInternal(src, pendingFile, false);
+    boolean skipReplicationCheck = false;
+    if (pendingFile.getStoragePolicyID() == HdfsConstants.DB_STORAGE_POLICY_ID){
+      skipReplicationCheck = true;
+    }
+
+    finalizeINodeFileUnderConstructionInternal(src, pendingFile, skipReplicationCheck);
   }
 
   private void finalizeINodeFileUnderConstructionStoredInDB(String src,
```

Comment:
shouldn't we remove this function if the above one handle the case of files in the DB?